pktfwd/manager.go

```diff
@@ -233,16 +233,25 @@ func (m *Manager) statusRoutine(bgCtx context.Context, errC chan error) {
 	}
 }
 
+func (m *Manager) networkRoutine(bgCtx context.Context, errC chan error) {
+	err := m.netClient.RefreshRoutine(bgCtx)
+	if err != nil {
+		errC <- errors.Wrap(err, "Couldn't refresh account server token")
+	}
+}
+
 func (m *Manager) startRoutines(bgCtx context.Context, err chan error, runTime time.Time) {
-	var errC = make(chan error)
+	var errC = make(chan error, 4)
```

Comment:
In the situation where `statusRoutine`, `networkRoutine` and `gpsRoutine` might all receive an error at the same time, they'll all try to pass it to `errC` at the same time, which will block these goroutines (since `errC` is of size 1 currently) and prevent them from receiving the `<-bgCtx.Done()` information. This means we won't be able to shut these goroutines gracefully. It would probably be better if we let those routines shoot their error, and then - since their cancel function will be called - let them shut gracefully.

You're also making me realize that [closing the channel](https://github.com/TheThingsNetwork/packet_forwarder/pull/34/files#diff-6001e964606e3a2a65dacbb3c8abed6bL252) will make the goroutines panic in that situation, so if that's good for you, I'll just remove this `close()`.